### PR TITLE
[enhancement] Nav-level increase when cloning sub-menus

### PR DESCRIPTION
**Pull request by petrklus**
Please find proposed amend to ResponsiveNav.js to better support multi-level responsive menus. When data-o-hierarchical-nav-is-cloneable is specified for a parent, the script ensures that the nav-level of the cloned submenu is increased to compensate for it being placed under the "More/Menu" elements.

### CODE REVIEW

**Comment by petrklus** [comment]
Demo: http://cnkpreview.s3.amazonaws.com/ft1_origami/o-hierarchical-nav/responsive-nav-cloneable.html

**Comment by petrklus** [comment]
However, while the above works - I would imagine there may be better ways of tagging the menus to either be preserved/flattened.

File: src/js/ResponsiveNav.js [hunk 1]
```diff
@@ -64,6 +64,10 @@ function ResponsiveNav(rootEl) {
 		cloneEl.removeAttribute('data-o-hierarchical-nav-is-cloneable');
 		// recurse through children and amend any id values to maintain uniqueness
 		prefixIds(el);
+        
+        // increase level of menus
+        incrementMenuDepths(cloneEl)
```
Comment (AlbertoElias) [inline src/js/ResponsiveNav.js]: indentation and semicolon missing

File: src/js/ResponsiveNav.js [hunk 2]
```diff
@@ -75,6 +79,27 @@ function ResponsiveNav(rootEl) {
 		}
 	}
 
+	function incrementMenuDepths(el) {
+		// id's are prefixed to ensure that any id based functionality uses the visible element
```
Comment (AlbertoElias) [inline src/js/ResponsiveNav.js]: what ids?

File: src/js/ResponsiveNav.js [hunk 3]
```diff
@@ -75,6 +79,27 @@ function ResponsiveNav(rootEl) {
...
+		// relates to as it uses the first matching id in the document
+		let child;
+		if (el.hasChildNodes()) {
+			const children = el.childNodes;
+			for (let i = 0, l = children.length; i < l; i++) {
+				child = children[i];
+				if (child instanceof HTMLElement) {
+					if (child.hasAttribute('data-o-hierarchical-nav-level')) {  
+                        // increment level
+                        let oldLevel = isNaN(parseInt(child.getAttribute('data-o-hierarchical-nav-level'))) ? 0 : parseInt(child.getAttribute('data-o-hierarchical-nav-level'))
```
Comment (AlbertoElias) [inline src/js/ResponsiveNav.js]: Indentation and might be worth taking this out to a separate variable: `parseInt(child.getAttribute('data-o-hierarchical-nav-level'))`

File: src/js/ResponsiveNav.js [hunk 4]
```diff
@@ -75,6 +79,27 @@ function ResponsiveNav(rootEl) {
...
+		let child;
+		if (el.hasChildNodes()) {
+			const children = el.childNodes;
+			for (let i = 0, l = children.length; i < l; i++) {
+				child = children[i];
+				if (child instanceof HTMLElement) {
+					if (child.hasAttribute('data-o-hierarchical-nav-level')) {  
+                        // increment level
+                        let oldLevel = isNaN(parseInt(child.getAttribute('data-o-hierarchical-nav-level'))) ? 0 : parseInt(child.getAttribute('data-o-hierarchical-nav-level'))              
+                        child.setAttribute('data-o-hierarchical-nav-level', 1+oldLevel);
```
Comment (AlbertoElias) [inline src/js/ResponsiveNav.js]: to follow the general coding style, could this be `oldLevel + 1`?

**Comment by AlbertoElias** [comment]
Thanks a lot @petrklus I left a few comments and I've got another question, do the nav levels go back to normal?

**Comment by petrklus** [comment]
Thank you very much for your comments @AlbertoElias! I have fixed the whitespace issues (I guess Atom's auto-detection is not as smart as it could be) and also updated the comments and added a variable for the nav depth. Regarding your other question - there should not be a need for the nav levels to go back, as we are only operating on cloned elements, not the originals.

File: src/js/ResponsiveNav.js [hunk 5]
```diff
@@ -64,10 +64,10 @@ function ResponsiveNav(rootEl) {
 		cloneEl.removeAttribute('data-o-hierarchical-nav-is-cloneable');
 		// recurse through children and amend any id values to maintain uniqueness
 		prefixIds(el);
-        
-        // increase level of menus
-        incrementMenuDepths(cloneEl)
-        
+
+    // increase level of nested menus
+    incrementMenuDepths(cloneEl);
```
Comment (AlbertoElias) [inline src/js/ResponsiveNav.js]: still doesn't seem properly indented

File: src/js/ResponsiveNav.js [hunk 6]
```diff
@@ -80,19 +80,20 @@ function ResponsiveNav(rootEl) {
...
 				child = children[i];
 				if (child instanceof HTMLElement) {
-					if (child.hasAttribute('data-o-hierarchical-nav-level')) {  
-                        // increment level
-                        let oldLevel = isNaN(parseInt(child.getAttribute('data-o-hierarchical-nav-level'))) ? 0 : parseInt(child.getAttribute('data-o-hierarchical-nav-level'))              
-                        child.setAttribute('data-o-hierarchical-nav-level', 1+oldLevel);
+					if (child.hasAttribute('data-o-hierarchical-nav-level')) {
+						// increment nav-level when attribute present
+						let origNavLevel = parseInt(child.getAttribute('data-o-hierarchical-nav-level'))
+            let updatedNavLevel = (isNaN(origNavLevel) ? 0 : origNavLevel) + 1
```
Comment (AlbertoElias) [inline src/js/ResponsiveNav.js]: indentation

**Comment by AlbertoElias** [comment]
Fair point, there´s still an indentation issue and you've also got merge conflicts. Thanks!

**Comment by petrklus** [comment]
Just had a look at the file and the indentation looks fine to me (https://github.com/cnk-digital-solutions/o-hierarchical-nav/blob/master/src/js/ResponsiveNav.js) - could you please confirm?

**Comment by AlbertoElias** [comment]
True! Just the merge conflicts then :)

**Comment by petrklus** [comment]
@AlbertoElias should be all there!

**Pull request merged**